library/Fission/Challenge/Types.hs

```diff
@@ -0,0 +1,59 @@
+module Fission.Challenge.Types (Challenge(..)) where
+  
+import           Database.Persist.Class
+import           Database.Persist.Types
+import           Database.Persist.Sql
+
+import qualified RIO.ByteString.Lazy as Lazy
+import qualified RIO.Text            as Text
+
+import qualified Network.IPFS.Internal.UTF8 as UTF8
+
+import           Data.Swagger
+import           Servant
+
+import           Fission.Prelude
+
+
+newtype Challenge = Challenge { unChallenge :: Text }
+  deriving          ( Generic )
+  deriving anyclass ( ToSchema
+                    , ToParamSchema
+                    )
```

Comment:
There's probably a few of these left in the code, but if possible let's do explicit instances for `ToSchema` and `ToParamSchema`. You can likely also drop the `Generic` once that's done.